src/lib/promise/js/promise.js

```diff
@@ -0,0 +1,147 @@
+// Provides: deferred_run
+function deferred_run(func) {
+  var deferred = {
+    promise: Promise.resolve()
+      .then(func) // the ocaml types don't know this, but func can actually be async or sync
```

Comment:
Can you restructure this to
```js
.then(function (value) {
    deferred.value = func(value);
    deferred.isDetermined = true;
    return deferred.value;
})
```
instead of using 2 `then` calls? Similarly elsewhere.

Reply:
I'd rather not, because then I lose the secret super-power that `f` can either return a Promise which holds `value` or `value` itself. `.then` "squashes" Promises if you will

Reply:
I think exactly this might be needed when I make some of the bindings.js functions async, to remove the spin lock